### DISCUSSION TITLE
What is "--no-fork" for brew bump-formula-pr used for?

### DISCUSSION THREAD

**Discussion by jsjoeio** [Homebrew/discussions#1532] [Tap maintenance and Homebrew development] [answered]
I wasn't sure what the docs meant [here](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/dev-cmd/bump-formula-pr.rb#L51-L52).

Is this something that only works if you have special access to `homebrew-core`?

Or, is it a special way to bump your formula without having to keep your fork of `hombrew-core` up-to-date?

🤞 hoping for the latter!

**Comment by gromgit** [comment, marked as answer]
Yes to the former, no to the latter. The PR process necessarily synchronizes your `homebrew/core` fork with your local core repo.

And why would you want to _NOT_ "keep your fork of `homebrew/core` up-to-date"? Letting it fall behind will eventually prevent you from issuing PRs, simply because your proposed diffs can no longer be matched to the mainline.

**Reply by jsjoeio** [reply]
Bummer. It's not that I don't want to keep it up-to-date. I just want it to stay up to date automatically 😂

As part of our release process, we run `bump-formula-pr` using a bot account. We have to figure out how to keep the bot's fork of `hombrew/core` up-to-date by fetching upstream and pushing before running `bump-formula-pr`.

I think we found a solution though but was curious if there was an easier way.

**Reply by carlocab** [reply]
`bump-formula-pr` should be running `brew update` before doing anything else (unless you have `HOMEBREW_NO_AUTO_UPDATE` set, or similar) so you shouldn't really need to worry about keeping the fork up-to-date.

**Reply by jsjoeio** [reply]
Oh! Interesting...

I believe we have a GitHub token in the environment which is used when running `bump-formula-pr`. (see [here](https://github.com/cdr/code-server/pull/3010/files)).

The first time it ran everything successfully.

The second time it ran, the fork was out-of-date (see [PR](https://github.com/Homebrew/homebrew-core/pull/77469#issuecomment-842645770)). Could this be that it didn't have the right permissions to update the fork or something?

**Reply by carlocab** [reply]
Maybe. GitHub Actions is voodoo. Could just not be interacting well with whatever cached version of Homebrew/core GitHub have set up in their CI instances.

I suggest using the [`setup-homebrew` action](https://github.com/Homebrew/actions/tree/master/setup-homebrew) to make sure whatever local repository you start with is up-to-date. Just add [this step](https://github.com/Homebrew/homebrew-core/blob/699b1f1b47524ff89d3c3dcac4155a34e8c0a7ae/.github/workflows/tests.yml#L22-L24) before your [`Configure git` step](https://github.com/cdr/code-server/blob/66cabab5785678967cbff6a058330fb36c5d7e77/.github/workflows/publish.yaml#L46-L49).

**Reply by jsjoeio** [reply]
Yes, it is voodoo 😅 

Ahh, didn't know about that. I'll give that a try. Thanks @carlocab 🙌

**Reply by tlk** [reply]
> `bump-formula-pr` should be running `brew update` before doing anything else (unless you have `HOMEBREW_NO_AUTO_UPDATE` set, or similar) so you shouldn't really need to worry about keeping the fork up-to-date.

..

> The first time it ran everything successfully.
> The second time it ran, the fork was out-of-date


Note that the default value for [HOMEBREW_AUTO_UPDATE_SECS](https://docs.brew.sh/Manpage#environment) used by the Homebrew auto-update mechanism is 300 seconds.

Perhaps [this logic](https://github.com/Homebrew/brew/blob/e4118bc4ca2bcb77dfd0bb1e823b6b8ff7f4b877/Library/Homebrew/brew.sh#L181-L193) applies to `bump-formula-pr` and can explain @jsjoeio's observations?
